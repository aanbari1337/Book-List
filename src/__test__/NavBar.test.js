import React from 'react'
import { render, unmountComponentAtNode } from "react-dom";
import { act } from 'react-dom/test-utils'
import Navbar from '../components/Navbar'
import renderer from 'react-test-renderer'
import { BookContext } from '../contexts/BookContext';

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it("renders with text content",()=>{
  const books = []
  act(()=>{
    render(
      <BookContext.Provider value={{books}}>
        <Navbar />
      </BookContext.Provider>, 
      container);
    })
    let child = document.getElementById("h1")
    expect(child.textContent).toBe("Book List")
})

it("matches snapshot",()=>{
  const books = []
  const tree = renderer.create(
      <BookContext.Provider value={{books}}>
        <Navbar />
      </BookContext.Provider>
    ).toJSON()
    expect(tree).toMatchSnapshot() 
})