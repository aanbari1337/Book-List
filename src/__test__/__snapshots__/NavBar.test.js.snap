// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="navbar"
>
  <h1
    id="h1"
  >
    Book List
  </h1>
  <p>
    Currently you have 
    0
     books to get through...
  </p>
</div>
`;
